staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go

```diff
@@ -83,6 +85,8 @@ type NewInformerFunc func({{.clientSetPackage|raw}}, {{.timeDuration|raw}}) cach
 type SharedInformerFactory interface {
 	Start(stopCh <-chan struct{})
 	InformerFor(obj {{.runtimeObject|raw}}, newFunc NewInformerFunc) {{.cacheSharedIndexInformer|raw}}
+	ExtraIndexers() {{.cacheIndexers|raw}}
+	KeyFunction() {{.cacheKeyFunc|raw}}
```

Comment:
One comment from Jordan that I didn't mention before (sorry!) was trying to avoid needing to do this. We should see if we can adjust the default informer functions in some way so this isn't needed.

Reply:
think I addressed it. I added variadic options to the NewInformerFunc type. I was unable to find a case of someone defining their own implementation of that function so as long as we don't take out the redundant resyncPeriod all callsites should remain unbroken. I could also create a parallel implementation with a `NewInformerFuncWithOptions` and `InformerForWithOptions` but it seemed a little heavy considering most of that is internal generated code.